crates/primitives/src/utils.rs

```diff
@@ -1,33 +1,32 @@
-use bitcoin::{
-    hashes::{sha256d, Hash},
-    Wtxid,
-};
 use serde::{Deserialize, Serialize};
 
-use crate::prelude::Buf32;
+use crate::{hash::sha256d, prelude::Buf32};
 
-/// Generates cohashes for an wtxid in particular index with in given slice of wtxids.
+/// Generates cohashes and computes the Merkle root for a transaction ID at a specific index
+/// within a given slice of elements that can be converted to `Buf32`.
+///
+/// This function supports any type that implements the `Into<Buf32>` trait, such as `txids` or
+/// `wtxids`.
 ///
 /// # Parameters
-/// - `wtxids`: The witness txids slice
-/// - `index`: The index of the txn for which we want the cohashes
+/// - `elements`: A slice of elements (`txids` or `wtxids`) that can be converted into `Buf32`.
+/// - `index`: The index of the transaction for which we want the cohashes.
 ///
 /// # Returns
-/// - A tuple `(Vec<Buf32>, Buf32)` containing the cohashes and the merkle root
+/// - A tuple `(Vec<Buf32>, Buf32)` containing the cohashes and the Merkle root.
 ///
 /// # Panics
```

Comment:
```suggestion
/// # Panics
///
```
empty line after headers